src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py

```diff
@@ -51,11 +51,11 @@
             "contents": {
                 # monthly et0 maps, each file ending in a number 1-12
                 "[MONTH]": {
+                    **spec_utils.ET0,
                     "about": _(
                         "Twelve files, one for each month. File names must "
                         "end with the month number (1-12)."),
```

Comment:
Is it only necessary that the filename end in a supported number? Or does it need to have a preceding underscore, like precip_1.tif? I was under the impression that the latter was required, but maybe not.

Reply:
The current version of the SWY source code only requires that the month index is there.  The underscore is optional.  So `foo1.tif` and `foo_1.tif` are both valid and accepted by the model.

I'm trying to construct a degenerate case, but the regular expression is pretty carefully constructed to make it hard to provide ambiguous filenames.

Reply:
Ok, good to know. It would be good to provide an example like "precip1.tif or precipitation_1.tif", but that might be fine to add after the ARGS_SPEC entry in the User Guide.

Reply:
@phargogh, my original thought was that examples would go in the UG so as to keep the `ARGS_SPEC` concise. 

I think ideally the `ARGS_SPEC` would have the minimum info necessary to create a correct input. So in this case, if the file names are matched with a regex, we would specify that regex (or a concise human-readable description of the regex pattern).

But as you point out, it's tricky to phrase a description of even a pretty simple regex. Cases like this make me think maybe it's better to require an exact file name rather than do pattern matching. Most people will likely have to rename their files anyway so that they each end with the correct month number. Is it really helpful to allow an arbitrary prefix? If we required that the files are named `evapotranspiration1.tif`, `evapotranspiration2.tif`, and so on, or `january.tif`, `february.tif`, etc., that's easier to document and easier for users to do correctly.

Reply:
Seems fine to have the ARGS_SPEC be minimalist, as long as it contains everything necessary. While I'm sure that it's easier on the coding side to require an exact file name, from a user point of view it would be annoyingly limiting. For example, it's common to run multiple scenarios with different climate change inputs for precip and ET0, and if all of those scenario files had to be named the same generic thing, it would be much more complicated to organize input data and know which scenario the different files represent just by looking at their names. So I strongly advocate for retaining flexibility for the rest of the file names.

Reply:
Ok thanks for weighing in @newtpatrol @emlys !  For the moment let's leave this functionality as-is.  I just resolved a merge conflict in `HISTORY`, so @emlys if you're OK with the addition of these couple of examples then I think this might be ready for one last look-over (and assuming the tests pass).